src/main/webapp/app/view/project/admin/locations/project-locations.component.ts

```diff
@@ -64,17 +68,29 @@ export class ProjectLocationsComponent implements OnInit, OnDestroy {
   }
 
   private loadLocations(): void {
-    this.locationService.findAllByProject(this.project!, { sort: ['name,asc'] }).subscribe((locations: HttpResponse<Location[]>) => {
-      this.loadedLocations = locations.body || [];
-
-      this.loadedLocations.forEach(location =>
-        this.sectionService
-          .findAllByLocation(location, { sort: ['name,asc'] })
-          .subscribe((sections: HttpResponse<Section[]>) => (location.sections = sections.body || []))
+    this.locationService
+      .findAllByProject(this.project!, { sort: ['name,asc'] })
+      .pipe(
+        map((response: HttpResponse<Location[]>) => response.body || []),
+        mergeMap(x => x),
+        map((location: Location) =>
+          forkJoin({
+            location: of(location),
+            sectionsResponse: this.sectionService.findAllByLocation(location, { sort: ['name,asc'] }),
+          })
+        ),
+        concatAll(),
+        map((value: { location: Location; sectionsResponse: HttpResponse<Section[]> }) => {
+          const location = value.location;
+          location.sections = value.sectionsResponse.body || [];
+          return location;
+        })
+      )
+      .subscribe(
+        (location: Location) => this.loadedLocations.push(location),
+        () => {},
+        () => (this.locations = this.loadedLocations)
```

Comment:
@Marcarrian I tried to remove this nested `subscribe` as well. can you please verify it is well done? I am not that used to `pipe`..

Reply:
Also du hast immer noch nested subscriptions.. Unter anderem wird diese Methode ja im constructor aufgerufen und dort im subscribe von dem router data observable.
Und dann zu deiner eigentlichen Frage: Es ist viel zu kompliziert. Warum ist das so? Ich verstehe gar nicht, was hier genau passiert. Werden hier nicht einfach alle Locations mit deren Sections geladen? Das wäre doch einfach ein query an das Backend und das wars?

Reply:
auch wahr. die obige habe ich mit `tap` entfernt.

Reply:
Ja viel besser 👍